src/mbgl/gl/gl_object_store.hpp

```diff
@@ -107,10 +108,11 @@ class TexturePoolHolder : private util::noncopyable {
     TexturePoolHolder() { ids.fill(0); }
     ~TexturePoolHolder() { reset(); }
 
-    TexturePoolHolder(TexturePoolHolder&& o) noexcept : ids(std::move(o.ids)) {}
-    TexturePoolHolder& operator=(TexturePoolHolder&& o) noexcept { ids = std::move(o.ids); return *this; }
+    TexturePoolHolder(TexturePoolHolder&& o) noexcept : ids(std::move(o.ids)), objectStore(o.objectStore) { o.ids.fill(0); }
+    TexturePoolHolder& operator=(TexturePoolHolder&& o) noexcept { ids = std::move(o.ids); objectStore = o.objectStore; return *this; }
 
-    explicit operator bool() { return std::none_of(ids.begin(), ids.end(), [](int id) { return id == 0; }); }
+    operator bool() = delete;
```

Comment:
There is no compiler-generated default `operator bool()`, so you don't need this.